audio_service/darwin/Classes/AudioServicePlugin.m

```diff
@@ -289,17 +293,40 @@ - (void) updateNowPlayingInfo {
             nowPlayingInfo[MPMediaItemPropertyArtist] = mediaItem[@"artist"];
         }
         if (mediaItem[@"duration"] != [NSNull null]) {
-            nowPlayingInfo[MPMediaItemPropertyPlaybackDuration] = [NSNumber numberWithLongLong: ([mediaItem[@"duration"] longLongValue] / 1000)];
+            nowPlayingInfo[MPMediaItemPropertyPlaybackDuration] = [NSNumber numberWithUnsignedLong: ([mediaItem[@"duration"] unsignedLongValue] / 1000)];
```

Comment:
I'm pretty sure this is wrong, my original code may also be wrong. The duration is signed and double (effectively). My code looks like it looses precision.